x-pack/plugins/cases/public/components/create/assignees.tsx

```diff
@@ -0,0 +1,148 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import React, { memo, useState } from 'react';
+import {
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFormRow,
+  EuiLink,
+  EuiSelectableListItem,
+  EuiTextColor,
+} from '@elastic/eui';
+import {
+  UserProfileWithAvatar,
+  UserAvatar,
+  getUserDisplayName,
+} from '@kbn/user-profile-components';
+import { UseField, FieldConfig } from '../../common/shared_imports';
+import { useSuggestUserProfiles } from '../../containers/user_profiles/use_suggest_user_profiles';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import { useGetCurrentUserProfile } from '../../containers/user_profiles/use_get_current_user_profile';
+import { OptionalFieldLabel } from './optional_field_label';
+import * as i18n from './translations';
+
+interface Props {
+  isLoading: boolean;
+}
+
+const getConfig = (): FieldConfig => ({
+  label: i18n.ASSIGNEES,
+  defaultValue: [],
+});
+
+const userProfileToComboBoxOption = (userProfile: UserProfileWithAvatar) => ({
+  label: getUserDisplayName(userProfile.user),
+  value: userProfile.uid,
+  user: userProfile.user,
+  data: userProfile.data,
+});
+
+const comboBoxOptionToAssignee = (option: EuiComboBoxOptionOption) => ({ uid: option.value });
+
+const AssigneesComponent: React.FC<Props> = ({ isLoading: isLoadingForm }) => {
+  const { owner } = useCasesContext();
+  const [searchTerm, setSearchTerm] = useState('');
+  const [selectedOptions, setSelectedOptions] = useState<EuiComboBoxOptionOption[]>();
+  const { data: currentUserProfile, isLoading: isLoadingCurrentUserProfile } =
+    useGetCurrentUserProfile();
+
+  const { data: userProfiles, isLoading: isLoadingSuggest } = useSuggestUserProfiles({
+    name: searchTerm,
+    owners: owner,
+  });
+
+  const options =
+    userProfiles?.map((userProfile) => userProfileToComboBoxOption(userProfile)) ?? [];
+
+  const onSearchComboChange = (value: string) => {
+    if (!isEmpty(value)) {
+      setSearchTerm(value);
+    }
+  };
+
+  const isLoading = isLoadingForm || isLoadingCurrentUserProfile || isLoadingSuggest;
+
+  const renderOption = (
+    option: EuiComboBoxOptionOption,
+    searchValue: string,
+    contentClassName: string
+  ) => {
+    const { user, data, value } = option as EuiComboBoxOptionOption<string> & UserProfileWithAvatar;
+
+    return (
+      <EuiSelectableListItem
+        key={value}
+        prepend={<UserAvatar user={user} avatar={data.avatar} size="s" />}
+        className={contentClassName}
+        append={<EuiTextColor color="subdued">{user.email}</EuiTextColor>}
+      >
+        {getUserDisplayName(user)}
+      </EuiSelectableListItem>
+    );
+  };
+
+  return (
+    <UseField path="assignees" config={getConfig()}>
+      {(field) => {
+        const { setValue } = field;
+
+        const onComboChange = (currentOptions: EuiComboBoxOptionOption[]) => {
```

Comment:
Correct, I forgot about it. Thanks!

Reply:
I get this linting error because the functions are inside another function:

<img width="687" alt="Screenshot 2022-09-02 at 3 19 30 PM" src="https://user-images.githubusercontent.com/7871006/188141459-7b02dbcc-a497-4538-9c25-f9c66e3d3626.png">

Reply:
Seems like we might be able to get around that by using the `component` prop of `UseField`:

[x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/other_type_name_parameter.tsx](https://github.com/elastic/kibana/blob/main/x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/other_type_name_parameter.tsx#L60-L85)

Maybe that's overkill though 🤷‍♂️

Reply:
Nice, I wasn't aware of that. I will take a look.

Reply:
Done!